### DISCUSSION TITLE
kratos project compiles incorrectly

### DISCUSSION THREAD

**Discussion by wangxingzhen** [apache/skywalking#11770] [Q&A] [answered]
When I build kratos, I get this error. My project has only grpc service, no http service. Is this the reason? There are more than a dozen microservices in this project directory, and I tried to add skywalking to one of them during build, but this error occurred.

go build -tags netgo -o service-market -toolexec "/skywalking-go/bin/skywalking-go-agent--darwin-arm64 -config /skywalking-go/tools/go-agent/config/agent.default.yaml" -a .

1 # github.com/go-kratos/kratos/v2/transport/http
/var/folders/fd/6tb4j71n3rj4tdx6scv9fj400000gp/T/go-build842865790/b278/skywalking_enhance_http_client_middleware.go:38:24: tr.RequestHeader().Add undefined (type transport.Header has no field or method Add)
2024/01/14 21:59:11 exit status 2

**Comment by wu-sheng** [comment]
I think you need to provide a demo codes to help to reproduce the error.

**Reply by wangxingzhen** [reply, marked as answer]
I know why!The reason is that my version of kratos is too low, my version is v2.5.3, and the latest version of kratos is v2.7.2. After I upgraded the version, the compilation was successful. Somewhat confusing, kratos v2.5.3, which was released on November 10th, 2022, is also not supported.

github.com/go-kratos/kratos/v2 v2.7.2
github.com/go-kratos/kratos/v2 v2.5.3

**Reply by wu-sheng** [reply]
That is possible, because the minor version is not a patch.
This depends on the maintaining mode of that project.
